pallets/ethereum-xcm/src/lib.rs

```diff
@@ -126,53 +128,85 @@ pub mod pallet {
 			xcm_transaction: EthereumXcmTransaction,
 		) -> DispatchResultWithPostInfo {
 			let source = T::XcmEthereumOrigin::ensure_origin(origin)?;
+			Self::validate_and_apply(source, xcm_transaction)
+		}
 
-			let (who, account_weight) = pallet_evm::Pallet::<T>::account_basic(&source);
-
-			let transaction: Option<Transaction> =
-				xcm_transaction.into_transaction_v2(U256::zero(), who.nonce);
-			if let Some(transaction) = transaction {
-				let transaction_data: TransactionData = (&transaction).into();
+		/// Xcm Transact an Ethereum transaction through proxy.
+		#[pallet::weight(<T as pallet_evm::Config>::GasWeightMapping::gas_to_weight({
+			match xcm_transaction {
```

Comment:
I would add the db read for the proxy here

Reply:
If we increase it, we'd break the assumption of 21000 gas base cost for a xcm-sent ethereum transfer, effectively becoming 22000 - compared with a Frame transfer, where if we do the weight to gas conversion it would be ~6400 gas. More over xcm side is already charging each instruction using a very conservative cost (8000 "gas" per instruction). All that makes me think the extra read overhead does not need to be accounted for, at least not in terms of how currently the fee costs are laid out.

Reply:
Will resolve this conversation, as we seem to have an agreement that pallet-ethereum will already over-estimate